crates/ark/src/modules/positron/graphics.R

```diff
@@ -134,107 +90,178 @@ renderWithPlotDevice <- function(filepath, format, width, height, res, type) {
     # Replace bindings.
     env_bind_force(baseenv(), ".Devices", .Devices)
     env_bind_force(baseenv(), ".Device", newDevice)
+
+    # Also set ourselves as a known interactive device.
+    # Used by `dev.interactive()`, which is used in `stats:::plot.lm()`
+    # to determine if `devAskNewPage(TRUE)` should be set to prompt before
+    # each new plot is drawn.
+    grDevices::deviceIsInteractive(name)
 }
 
-# Create a snapshot of the current plot.
+# Create a recording of the current plot.
 #
-# This saves the plot's display list, so it can be used
-# to re-render plots as necessary.
+# This saves the plot's display list, so it can be used to re-render plots as
+# necessary.
 #' @export
-.ps.graphics.createSnapshot <- function(id) {
-    # Flush any pending plot actions.
-    grDevices::dev.set(grDevices::dev.cur())
-    grDevices::dev.flush()
-
-    # Create the plot snapshot.
-    recordedPlot <- grDevices::recordPlot()
-
-    # Get the path to the plot snapshot file.
-    snapshotPath <- .ps.graphics.plotSnapshotPath(id)
+.ps.graphics.recordPlot <- function(id) {
+    # Create the plot recording
+    recording <- grDevices::recordPlot()
 
-    # Save it to disk.
-    saveRDS(recordedPlot, file = snapshotPath)
+    # Add the recording to the persistent list
+    addRecording(id, recording)
 
-    # Return the path to that snapshot file.
-    snapshotPath
+    invisible(NULL)
 }
 
 #' @export
-.ps.graphics.renderPlot <- function(id, width, height, dpr, format) {
-    # If we have an existing snapshot, render from that file.
-    snapshotPath <- .ps.graphics.plotSnapshotPath(id)
-    if (file.exists(snapshotPath)) {
-        .ps.graphics.renderPlotFromSnapshot(id, width, height, dpr, format)
-    } else {
-        .ps.graphics.renderPlotFromCurrentDevice(id, width, height, dpr, format)
-    }
```

Comment:
No more reliance on this if/else. When we render we unconditionally pull it from a pre-recorded snapshot now.